pkg/cloudclient/aws/private.go

```diff
@@ -62,45 +62,50 @@ var (
 	userdataEndVerifier   string = "USERDATA END"
 )
 
-func newClient(ctx context.Context, logger ocmlog.Logger, accessID, accessSecret, sessiontoken, region,
-	instanceType string, tags map[string]string, profile string) (*Client, error) {
+func getEc2ClientFromInput(input ClientInput) (*ec2.Client, error) {
 	var cfg aws.Config
 	var err error
-	if profile != "" {
-		cfg, err = config.LoadDefaultConfig(ctx,
-			config.WithSharedConfigProfile(profile),
-			config.WithRegion(region),
-		)
-	} else {
-		cfg, err = config.LoadDefaultConfig(ctx,
-			config.WithRegion(region),
-			config.WithCredentialsProvider(credentials.StaticCredentialsProvider{
-				Value: aws.Credentials{
-					AccessKeyID: accessID, SecretAccessKey: accessSecret, SessionToken: sessiontoken,
-				},
-			}),
-		)
+
+	cfg, err = config.LoadDefaultConfig(input.ExecConfig.Ctx,
```

Comment:
I'm seeing we're removing `if profile != "" {` check so how do we create cfg based on AWS profile?